api/envoy/api/v2/listener/quic_config.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+
+package envoy.api.v2.listener;
+
+option java_outer_classname = "ListenerProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.api.v2.listener";
+option csharp_namespace = "Envoy.Api.V2.ListenerNS";
+option ruby_package = "Envoy::Api::V2::ListenerNS";
+
+// Configuration specific to the QUIC protocol.
+// Next id: 4
+message QuicConfigProto {
```

Comment:
Now that QUIC is a transport and H3 is an application, separate configs make sense, though it's not how the code is currently structured.

If we create separate configs, I'm not sure whether nesting makes more sense or putting them side-by-side? If TcpProtocolOptions is inside Http2ProtocolOptions, then following that pattern is the most obvious approach.

Reply:
Http2ProtocolOptions is in protocol.proto so we can easily reuse it, but don't we have issues with HPACK settings? Should we make QuicProtocolOptions and Http3ProtocolOptions as needed and clean this up in v3 if there are shared message possibilities?

Reply:
Good point that Http2ProtocolOptions has unwanted HPACK setting config except for the stream multiplexing config which we wanted in QuicProtocolOptions. And Http3ProtocolOptions will have its own config for things like QPACK table size. I think it's better not to reuse Http2ProtocolOptions, instead, adding all the fields needed in our own protobuf messages, basically, as it is currently defined now. WDYT?

Since I didn't know protocol.proto before, would it make more sense to define QuicProtocolOptions and Http3ProtocolOptions there instead?